Unexpected Panic error in subfinder

Hell when I was testing performance subfinder each resources I faced panic error and the full error was : 
```code
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd57b8a]

goroutine 66 [running]:
github.com/projectdiscovery/ratelimit.(*MultiLimiter).Stop(0x56240f?, {0x0?, 0xc000e1be01?, 0x16a75a0?})
	/home/runner/go/pkg/mod/github.com/projectdiscovery/ratelimit@v0.0.26/keyratelimit.go:106 +0x2a
github.com/projectdiscovery/subfinder/v2/pkg/subscraping.(*Session).Close(0xc000386780)
	/home/runner/work/subfinder/subfinder/v2/pkg/subscraping/agent.go:128 +0x26
github.com/projectdiscovery/subfinder/v2/pkg/passive.(*Agent).EnumerateSubdomainsWithCtx.func1()
	/home/runner/work/subfinder/subfinder/v2/pkg/passive/passive.go:77 +0x60e
created by github.com/projectdiscovery/subfinder/v2/pkg/passive.(*Agent).EnumerateSubdomainsWithCtx in goroutine 1
	/home/runner/work/subfinder/subfinder/v2/pkg/passive/passive.go:37 +0x185

```
and also Im not well in golang but when I tested the performance with my [tool](https://github.com/sanjai-AK47/Subdominator) with subfinder and specially with resources I faced this error and after researching about this error I think it occurs when it try to read or write to the unallocated memory address and to analyze and handle this error I think the recover() built-in Go function will be helpfull and Im also attaching a screenshot about the error :

![image](https://github.com/projectdiscovery/subfinder/assets/119435129/48787fa4-8632-4b1b-ba06-94fbb4a2b0b1)

Kind regards,
[MR.Z](https://github.com/sanjai-AK47)

--- a/v2/pkg/passive/sources.go
+++ b/v2/pkg/passive/sources.go
@@ -125,7 +125,7 @@ func New(sourceNames, excludedSourceNames []string, useAllSources, useSourcesSup
 		if len(sourceNames) > 0 {
 			for _, source := range sourceNames {
 				if NameSourceMap[source] == nil {
-					gologger.Warning().Msgf("There is no source with the name: %s", source)
+					gologger.Fatal().Msgf("There is no source with the name: %s", source)
 				} else {
 					sources[source] = NameSourceMap[source]
 				}

--- a/v2/pkg/passive/sources_w_auth_test.go
+++ b/v2/pkg/passive/sources_w_auth_test.go
@@ -47,7 +47,12 @@ func TestSourcesWithKeys(t *testing.T) {
 			continue
 		}
 
-		apiKey := os.Getenv(fmt.Sprintf("%s_API_KEY", strings.ToUpper(source.Name())))
+		var apiKey string
+		if source.Name() == "chaos" {
+			apiKey = os.Getenv("PDCP_API_KEY")
+		} else {
+			apiKey = os.Getenv(fmt.Sprintf("%s_API_KEY", strings.ToUpper(source.Name())))
+		}
 		if apiKey == "" {
 			fmt.Printf("Skipping %s as no API key is provided\n", source.Name())
 			continue

